File: realm-common/src/main/java/io/github/solusmods/eternalcore/realm/EternalCoreRealm.java
package io.github.solusmods.eternalcore.realm;

import io.github.solusmods.eternalcore.realm.impl.RealmRegistry;
import io.github.solusmods.eternalcore.realm.impl.RealmStorage;
import io.github.solusmods.eternalcore.realm.impl.network.RealmNetwork;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;
import net.minecraft.resources.ResourceLocation;

import static io.github.solusmods.eternalcore.realm.ModuleConstants.MOD_ID;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class EternalCoreRealm {

    public static ResourceLocation create(String path) {
        return ResourceLocation.fromNamespaceAndPath(MOD_ID, path);
    }

    public static void init(){
        RealmStorage.init();
        RealmRegistry.init();
        RealmNetwork.init();
    }
}

File: realm-common/src/main/kotlin/io/github/solusmods/eternalcore/realm/EternalCoreRealm.kt
package io.github.solusmods.eternalcore.realm

import io.github.solusmods.eternalcore.realm.impl.RealmRegistry
import io.github.solusmods.eternalcore.realm.impl.RealmStorage
import io.github.solusmods.eternalcore.realm.impl.network.RealmNetwork
import net.minecraft.resources.ResourceLocation

object EternalCoreRealm {
    fun create(path: String): ResourceLocation {
        return ResourceLocation.fromNamespaceAndPath(ModuleConstants.MOD_ID, path)
    }

    @JvmStatic
    fun init() {
        RealmStorage.Companion.init()
        RealmRegistry.init()
        RealmNetwork.init()
    }
}
